### PR TITLE
[Suffix] modify ApplicantData and TI service to add suffix to client info

### PR DESCRIPTION
**Pull request by azhangggggg**
### Description

Modified ApplicantData to take `nameSuffix`,  and TI service to enabled adding suffix to client info. 
Added and updated unit tests on these services.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | under-development | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers)
- [x] Added an additional reviewer from outside your organization as FYI (if the primary reviewer is in the same organization as you)
- [x] Created unit and/or browser tests which fail without the change (if possible)

### Issue(s) this completes

Part of #8267

### CODE REVIEW

File: server/test/repository/AccountRepositoryTest.java [hunk 1]
```diff
@@ -75,32 +76,35 @@ public void updateClientNameTest_AllNonEmpty() {
...
     assertThat(applicantUpdateTest.getApplicantData().getApplicantMiddleName().get())
         .isEqualTo("James");
     assertThat(applicantUpdateTest.getApplicantData().getApplicantLastName().get())
         .isEqualTo("John");
+    assertThat(applicantUpdateTest.getApplicantData().getApplicantNameSuffix().get())
+        .isEqualTo("Jr.");
   }
 
   @Test
   public void updateClientNameTest_EmptyMiddleAndLastName() {
```
Comment (dkatzz) [inline server/test/repository/AccountRepositoryTest.java:91]: **suggestion:** Let's update the test name to indicate suffix is empty too?

File: server/test/services/applicant/ApplicantDataTest.java [hunk 2]
```diff
@@ -69,13 +69,30 @@ public void getApplicantName_fallsBackToWKPWhenNameIsEmail() {
...
     data.setUserName("First Middle Last");
     assertThat(data.getApplicantName()).isEqualTo(Optional.of("Last, First"));
     assertThat(data.getApplicant().getMiddleName().get()).isEqualTo("Middle");
   }
 
+  @Test
+  public void getApplicantName_withNameSuffixWithoutMiddleName_exist() {
+    ApplicantData data = createNewApplicantData();
+    data.setUserName("First Last Jr.");
+    assertThat(data.getApplicantName()).isEqualTo(Optional.of("Last, First"));
```
Comment (dkatzz) [inline server/test/services/applicant/ApplicantDataTest.java:83]: **question:** Outside the scope of this PR, but have we confirmed whether the suffix should be included in this type of applicant name? Like should it be "Last, First, Jr."? https://english.stackexchange.com/questions/116856/where-do-you-put-the-suffix-when-listing-the-last-name-first

**Review by dkatzz — APPROVED** [review]

File: server/test/services/applicant/ApplicantDataTest.java (hunk 2 above)
Comment (azhangggggg) [reply server/test/services/applicant/ApplicantDataTest.java:83]: Good point, I will raise it to shreya and/or sijia.

File: server/test/repository/AccountRepositoryTest.java (hunk 1 above)
Comment (azhangggggg) [reply server/test/repository/AccountRepositoryTest.java]: yup updated test name.

**Pull request merged**